### PR TITLE
Add ability to omit keysets in superpack payload

### PR DESCRIPTION
**Pull request by disnet**

### CODE REVIEW

File: src/encoder.js [hunk 1]
```diff
@@ -346,15 +340,36 @@ function encodeValue(value: any, target: Array<number | string>) {
...
+function findIndex(iterable, predicate) {
+  let index = 0;
+  for (let a of iterable) {
+    if (predicate(a)) return index;
+    ++index;
+  }
+  return -1;
+}
+
+function isSortedArrayOfStringsSameAsSortedArrayOfStrings(a, b) {
```
Comment (michaelficarra) [inline src/encoder.js]: `isSortedArrayOfThingsSameAsSortedArrayOfThings`

File: src/encoder.js [hunk 2]
```diff
@@ -346,15 +340,36 @@ function encodeValue(value: any, target: Array<number | string>) {
   return target;
 }
 
-export default function encode(value: any): Array<number> {
+function findIndex(iterable, predicate) {
+  let index = 0;
+  for (let a of iterable) {
```
Comment (michaelficarra) [inline src/encoder.js]: This is going to depend on `Symbol.iterator` at runtime. We can't do that and remain IE 9 compatible.

File: test/spec/encoding.spec.js [hunk 3]
```diff
@@ -19,7 +19,8 @@ describe('encoding', function () {
     describe(category, function () {
       cases[category].forEach(function (t) {
         it('should encode ' + t.desc, function () {
-          expect(encode(t.value)).to.eql(t.bytes);
+          let opt = t.keysetsToOmit != null ? { keysetsToOmit: t.keysetsToOmit } : void 0;
```
Comment (michaelficarra) [inline test/spec/encoding.spec.js]: Use `== null` and flip the consequent/alternate.

File: src/encoder.js (hunk 2 above)
Comment (disnet) [inline src/encoder.js]: should we add the `for of` babel transform?

Comment (disnet) [inline src/encoder.js]: nvm, I'll just rewrite it

**Comment by michaelficarra** [comment]
LGTM once we get BrowserStack working. :cry:

**Pull request merged**